test/cases/generation.sh

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/bash
+set -euxo pipefail
+
+# Provision the software under test.
+/usr/libexec/osbuild-composer-test/provision.sh
+
+# Get OS data
+source /usr/libexec/osbuild-composer-test/set-env-variables.sh
+
+# Colorful timestamped output.
+function greenprint {
+    echo -e "\033[1;32m[$(date -Isecond)] ${1}\033[0m"
+}
+
+# install requirements
+sudo dnf -q -y install make \
+    git \
+    go \
+    ranger \
+    openssl \
+    rpm-build \
+    aws \
+    qemu \
+    setools-console \
+    setroubleshoot \
+    container-selinux \
+    lvm2
+
+# stop dnf-json socket
+sudo systemctl stop osbuild-dnf-json.socket
+
+# test the test cases generation
+WORKDIR=$(mktemp -d)
+mkdir -p "$WORKDIR"
+
+# test the test case generation when dnf-json socket is stopped
+sudo ./tools/test-case-generators/generate-test-cases\
```

Comment:
You should add
```
OSBUILD_LABEL=$(matchpathcon -n $(which osbuild))
chcon $OSBUILD_LABEL tools/image-info
```
before calling the generator script.

For more information, check the "important note" in https://github.com/osbuild/osbuild-composer/tree/main/test#image-tests